python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1299 @@
+
+    The TARTS system includes three main components:
+    - Wavenet: Estimates Zernike coefficients from donut images
+    - Alignnet: Handles image alignment and preprocessing
+    - Aggregatornet: Combines results from multiple donuts
+
+    Detector Support
+    ---------------
+    This task only supports corner wavefront sensors with detector names ending
+    with '_SW0' (extra-focal) or '_SW1' (intra-focal). Full array mode is not
+    currently supported.
+    """
+
+    # Class constants for processing
+    EXPECTED_IMAGE_DIMENSIONS = 3  # Expected dimensions for TARTS output
+    LOG_PRECISION = 3  # Decimal precision for logging Zernike values
+
+    ConfigClass = CalcZernikesNeuralTaskConfig
+    _DefaultName = "calcZernikesNeuralTask"
+
+    # Sentinel values for missing data
+    UNKNOWN_STRING = "Unknown"
+    UNKNOWN_INT = -1
+    UNKNOWN_FLOAT = float("nan")
+    UNKNOWN_ANGLE = 0.0
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the neural network-based Zernike estimation task.
+
+        This method sets up the TARTS neural network system, configures the
+        device (CPU or CUDA), and initializes the Noll indices for Zernike
+        coefficients. The TARTS system is loaded with the specified model
+        weights and dataset parameters from the configuration.
+
+        Parameters
+        ----------
+        **kwargs : Any
+            Additional keyword arguments passed to the parent PipelineTask
+            class.
+
+        Notes
+        -----
+        The initialization process:
+        1. Calls parent class constructor with super().__init__(**kwargs)
+        2. Sets up Noll indices from configuration
+        3. Validates configuration parameters
+        4. Loads TARTS neural network models with specified weights
+        5. Sets models to evaluation mode with .eval()
+        6. Extracts crop size from TARTS system
+        7. Configures device (CPU/CUDA) and moves models accordingly
+
+        Noll indices are configurable via the config.nollIndices parameter.
+        The default excludes Z1-Z3 (piston, tip, tilt) as these are typically
+        not measured in wavefront sensing.
+        """
+        super().__init__(**kwargs)
+
+        # Type annotation for mypy to understand the config structure
+        self.config: CalcZernikesNeuralTaskConfig
+
+        # Define default Noll indices to be used for wavefront estimation
+        self.nollIndices = self.config.nollIndices
+        self.log.debug("Configured Noll indices: %s", self.nollIndices)
+
+        # Deferred import of TARTS to handle cases where it's not in the build
+        try:
+            from tarts import NeuralActiveOpticsSys
+        except ImportError as e:
+            raise ImportError(
+                "TARTS neural network system not available. "
+                "Please ensure TARTS is installed and available in the Python path. "
+                f"Original error: {e}"
+            ) from e
```

Comment:
Aside from the comment above, there's minimal point in catching and re-raising the same error - I'd just leave without a `try` tbh.